library/core/src/main/java/com/google/android/exoplayer2/source/UnreportedDiscontinuityException.java

```diff
@@ -0,0 +1,26 @@
+package com.google.android.exoplayer2.source;
+
+import android.net.Uri;
+import com.google.android.exoplayer2.C;
+
+/**
+ * Thrown from the loader thread when an attempt is made to commit a sample that is far
+ * deviant from the expected sequence of timestamps in the SampleStream.   This is likely
+ * caused by a discontinuity in a segment that was not split and reported by metadata in
+ * an HLS (EXT-X-DISCONTINUITY) or DASH stream.
+ */
+public class UnreportedDiscontinuityException extends RuntimeException {
+
+  public final long timesUs;
+
+  /**
+   * Consturct the exception
+   *
+   * @param timesUs last timestamp before attempted commit of the deviant sample
```

Comment:
It would be nice to capture both timestamps: `lastSampleTimeUs` and `nextSampleTimeUs`. And to pass the whole `DataSpec` rather than just the `Uri`.

(Note that some HLS streams will use the same `Uri` for every segment, so at least knowing the range is pretty essential)

Reply:
Excellent idea.  Yes, `lastSampleTimeUs` will give us an idea how far into the issue occurs segment it is (`lastSampleTimeUs` - `lastSampleTimeUs`).